crypto/store/store_result.c

```diff
@@ -304,16 +279,17 @@ static EVP_PKEY *try_key_value_legacy(struct extracted_param_data_st *data,
     const unsigned char *der = data->octet_data, *derp;
     long der_len = (long)data->octet_data_size;
 
-    SET_ERR_MARK();
     /* Try PUBKEY first, that's a real easy target */
     if (ctx->expected_type == 0
         || ctx->expected_type == OSSL_STORE_INFO_PUBKEY) {
+        /* Discard DER decoding errors, it only means we return "empty handed" */
+        ERR_set_mark();
```

Comment:
IMO if the expected_type is set, we should not discard the errors.

Reply:
I have nothing against that thought, let's just see what happens with the tests...